backend/tilavarauspalvelu/api/rest/views.py

```diff
@@ -580,7 +580,7 @@ def robot_test_data_create_view(request: WSGIRequest) -> HttpResponse:
         cache.set(key=lock_key, value=True, timeout=None)
 
         try:
-            create_robot_test_data()
+            create_robot_test_data_task.delay()
```

Comment:
The lock_key removal (`cache.delete`) needs to be moved to the task logic from the finally block here.